src/serac/numerics/functional/detail/segment_L2.inl

```diff
@@ -178,8 +178,12 @@ struct finite_element<mfem::Geometry::SEGMENT, L2<p, c> > {
         f_buffer_type flux;
 
         for (int qx = 0; qx < q; qx++) {
-          source(qx) = reinterpret_cast<const double*>(&get<SOURCE>(qf_output[qx]))[i * ntrial + j];
-          flux(qx)   = reinterpret_cast<const double*>(&get<FLUX>(qf_output[qx]))[i * ntrial + j];
+          if constexpr (!is_zero<source_type>{}) {

```

Comment:
Should this be outside of the for loop?

Reply:
The behavior is equivalent both ways. I prefer this implementation slightly because it's a couple lines of code shorter.

Reply:
Also is `source` and `flux` being initialized to a reasonable value since its skipping this part and then used on line 189?

Reply:
> Is q zero when the source_type and/or flux_type is is_zero?

q is always greater than zero

> Also is source and flux being initialized to a reasonable value since its skipping this part and then used on line 189?

the initial value doesn't matter since it's overwritten in the for loop, or it is a `serac::zero` type which has no data in the first place